### ISSUE TITLE
Error in data-container

### ISSUE THREAD

**Issue by avdata99** [okfn/ckanext-unhcr#712]
The data container (created yesterday) give us 500 error 
https://ridl.unhcr.org/data-container/innovation-service

When we tried to add another user to it, it also give un an internal error.

I also see two similarly named containers in the dropdown on the Manage Membership page, not sure why... 

![image](https://user-images.githubusercontent.com/3237309/135872525-1f1e36ad-fcbf-4f10-ab44-d9bc70626c73.png)

### Log error

<pre>
| 1633360469472 | Error - <type 'exceptions.RuntimeError'>: maximum recursion depth exceeded while calling a Python object 
| 1633360469485 | URL: https://ridl.unhcr.org/data-container/innovation-service 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/weberror/errormiddleware.py', line 171 in __call__ 
| 1633360469485 | app_iter = self.application(environ, sr_checker) 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/webob/dec.py', line 147 in __call__ 
| 1633360469485 | resp = self.call_func(req, *args, **self.kwargs) 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/webob/dec.py', line 208 in call_func 
| 1633360469485 | return self.func(req, *args, **kwargs) 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/fanstatic/publisher.py', line 234 in __call__ 
| 1633360469485 | return request.get_response(self.app) |
| 1633360469485 | File '/usr/lib/python2.7/site-packages/webob/request.py', line 1053 in get_response |
| 1633360469485 | application, catch_exc_info=False) 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/webob/request.py', line 1022 in call_application 
| 1633360469485 | app_iter = application(self.environ, start_response) 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/webob/dec.py', line 147 in __call__ 
| 1633360469485 | resp = self.call_func(req, *args, **self.kwargs) 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/webob/dec.py', line 208 in call_func 
| 1633360469485 | return self.func(req, *args, **kwargs) 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/fanstatic/injector.py', line 54 in __call__ |
| 1633360469485 | response = request.get_response(self.app) 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/webob/request.py', line 1053 in get_response |
| 1633360469485 | application, catch_exc_info=False) 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/webob/request.py', line 1022 in call_application 
| 1633360469485 | app_iter = application(self.environ, start_response) 
| 1633360469485 | File '/srv/app/src/ckan/ckan/config/middleware/pylons_app.py', line 264 in inner 
| 1633360469485 | result = application(environ, start_response) 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/beaker/middleware.py', line 73 in __call__ 
| 1633360469485 | return self.app(environ, start_response) 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/beaker/middleware.py', line 156 in __call__ |
| 1633360469485 | return self.wrap_app(environ, session_start_response) 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/routes/middleware.py', line 131 in __call__ |
| 1633360469485 | response = self.app(environ, start_response) 
| 1633360469485 | File '/srv/app/src/ckan/ckan/config/middleware/common_middleware.py', line 33 in __call__ 
| 1633360469485 | return self.app(environ, start_response) 
| 1633360469485 | File '/srv/app/src/ckan/ckan/config/middleware/common_middleware.py', line 59 in __call__ 
| 1633360469485 | return self.app(environ, start_response) 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/pylons/wsgiapp.py', line 125 in __call__ 
| 1633360469485 | response = self.dispatch(controller, environ, start_response) 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/pylons/wsgiapp.py', line 324 in dispatch 
| 1633360469485 | return controller(environ, start_response) 
| 1633360469485 | File '/srv/app/src/ckan/ckan/lib/base.py', line 242 in __call__ 
| 1633360469485 | res = WSGIController.__call__(self, environ, start_response) 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/pylons/controllers/core.py', line 221 in __call__ 
| 1633360469485 | response = self._dispatch_call() 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/pylons/controllers/core.py', line 172 in _dispatch_call 
| 1633360469485 | response = self._inspect_call(func) 
| 1633360469485 | File '/usr/lib/python2.7/site-packages/pylons/controllers/core.py', line 107 in _inspect_call |
| 1633360469486 | result = self._perform_call(func, args) 
| 1633360469486 | File '/usr/lib/python2.7/site-packages/pylons/controllers/core.py', line 60 in _perform_call |
| 1633360469486 | return func(**args) 
| 1633360469486 | File '/srv/app/src/ckanext-unhcr/ckanext/unhcr/controllers/extended_organization.py', line 15 in read 
| 1633360469486 | return super(ExtendedOrganizationController, self).read(id) 
| 1633360469486 | File '/srv/app/src/ckan/ckan/controllers/group.py', line 223 in read 
| 1633360469486 | c.group_dict = self._action('group_show')(context, data_dict) 
| 1633360469486 | File '/srv/app/src/ckan/ckan/logic/__init__.py', line 471 in wrapped 
| 1633360469486 | result = _action(context, data_dict, **kw) 
| 1633360469486 | File '/srv/app/src/ckan/ckan/logic/action/get.py', line 1340 in organization_show 
| 1633360469486 | return _group_or_org_show(context, data_dict, is_org=True) 
| 1633360469486 | File '/srv/app/src/ckan/ckan/logic/action/get.py', line 1239 in _group_or_org_show |

<b> This happens houndred times

| 1633360469486 | include_users=include_users,) 
| 1633360469486 | File '/srv/app/src/ckan/ckan/lib/dictization/model_dictize.py', line 435 in group_dictize 
| 1633360469486 | context, include_groups=True) 
| 1633360469488 File '/srv/app/src/ckan/ckan/lib/dictization/model_dictize.py', line 55 in group_list_dictize 
| 1633360469488 for group, capacity in obj_list] 
</b>

| 1633360469488 File '/srv/app/src/ckan/ckan/lib/dictization/model_dictize.py', line 435 in group_dictize 
| 1633360469488 context, include_groups=True) 
| 1633360469488 File '/srv/app/src/ckan/ckan/lib/dictization/model_dictize.py', line 55 in group_list_dictize 
| 1633360469488 for group, capacity in obj_list] 
| 1633360469488 File '/srv/app/src/ckan/ckan/lib/dictization/model_dictize.py', line 435 in group_dictize 
| 1633360469488 context, include_groups=True) 
| 1633360469488 File '/srv/app/src/ckan/ckan/lib/dictization/model_dictize.py', line 55 in group_list_dictize 
| 1633360469488 for group, capacity in obj_list] 
| 1633360469488 File '/srv/app/src/ckan/ckan/lib/dictization/model_dictize.py', line 435 in group_dictize 
| 1633360469488 context, include_groups=True) 
| 1633360469488 File '/srv/app/src/ckan/ckan/lib/dictization/model_dictize.py', line 55 in group_list_dictize 
| 1633360469488 for group, capacity in obj_list] 

| 1633360469488 File '/srv/app/src/ckan/ckan/lib/dictization/model_dictize.py', line 434 in group_dictize 
| 1633360469488 _get_members(context, group, 'groups'), 
| 1633360469488 File '/srv/app/src/ckan/ckan/lib/dictization/model_dictize.py', line 329 in _get_members 
| 1633360469488 return q.all() 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/orm/query.py', line 2703 in all 
| 1633360469488 return list(self) 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/orm/query.py', line 2855 in __iter__ 
| 1633360469488 return self._execute_and_instances(context) 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/orm/query.py', line 2878 in _execute_and_instances 
| 1633360469488 result = conn.execute(querycontext.statement, self._params) 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/engine/base.py', line 945 in execute 
| 1633360469488 return meth(self, multiparams, params) 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/elements.py', line 263 in _execute_on_connection 
| 1633360469488 return connection._execute_clauseelement(self, multiparams, params) 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/engine/base.py', line 1046 in _execute_clauseelement 
| 1633360469488 if not self.schema_for_object.is_default else None) 
| 1633360469488 File '<string>', line 1 in <lambda> 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/elements.py', line 436 in compile 
| 1633360469488 return self._compiler(dialect, bind=bind, **kw) 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/elements.py', line 442 in _compiler 
| 1633360469488 return dialect.statement_compiler(dialect, self, **kw) 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py', line 435 in __init__ 
| 1633360469488 Compiled.__init__(self, dialect, statement, **kwargs) 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py', line 216 in __init__ 
| 1633360469488 self.string = self.process(self.statement, **compile_kwargs) 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py', line 242 in process 
| 1633360469488 return obj._compiler_dispatch(self, **kwargs) 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py', line 81 in _compiler_dispatch 
| 1633360469488 return meth(self, **kw) 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py', line 1738 in visit_select 
| 1633360469488 text, select, inner_columns, froms, byfrom, kwargs) 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py', line 1822 in _compose_select_body 
| 1633360469488 t = select._whereclause._compiler_dispatch(self, **kwargs) 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py', line 93 in _compiler_dispatch 
| 1633360469488 return meth(self, **kw) 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py', line 807 in visit_clauselist 
| 1633360469488 for c in clauselist.clauses) 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py', line 804 in <genexpr> 
| 1633360469488 s for s in 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py', line 807 in <genexpr> 
| 1633360469488 for c in clauselist.clauses) 
| 1633360469488 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py', line 93 in _compiler_dispatch 
| 1633360469488 return meth(self, **kw) 
| 1633360469489 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py', line 807 in visit_clauselist 
| 1633360469489 for c in clauselist.clauses) 
| 1633360469489 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py', line 804 in <genexpr> 
| 1633360469489 s for s in 
| 1633360469489 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py', line 807 in <genexpr> 
| 1633360469489 for c in clauselist.clauses) 
| 1633360469489 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py', line 93 in _compiler_dispatch 
| 1633360469489 return meth(self, **kw) 
| 1633360469489 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py', line 1025 in visit_binary 
| 1633360469489 return self._generate_generic_binary(binary, opstring, **kw) 
| 1633360469489 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py', line 1050 in _generate_generic_binary 
| 1633360469489 self, eager_grouping=eager_grouping, **kw) 
| 1633360469489 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py', line 93 in _compiler_dispatch 
| 1633360469489 return meth(self, **kw) 
| 1633360469489 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py', line 1183 in visit_bindparam 
| 1633360469489 name = self._truncate_bindparam(bindparam) 
| 1633360469489 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py', line 1239 in _truncate_bindparam 
| 1633360469489 bind_name = self._truncated_identifier("bindparam", bind_name) 
| 1633360469489 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py', line 1250 in _truncated_identifier 
| 1633360469489 anonname = name.apply_map(self.anon_map) 
| 1633360469489 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/elements.py', line 4079 in apply_map 
| 1633360469489 return self % map_ 
| 1633360469489 File '/usr/lib/python2.7/site-packages/sqlalchemy/util/_collections.py', line 729 in __missing__ 
| 1633360469489 self[key] = val = self.creator(key) 
| 1633360469489 File '/usr/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py', line 1269 in _process_anon 
| 1633360469489 return derived + "_" + str(anonymous_counter) 
| 1633360469489 RuntimeError: maximum recursion depth exceeded while calling a Python object 
| 1633360469489 CGI Variables 
| 1633360469489 ------------- 
| 1633360469489 AUTH_TYPE: 'cookie' 
| 1633360469489 CKAN_CURRENT_URL: '/data-container/innovation-service' 
| 1633360469489 CKAN_LANG: 'en' 
| 1633360469489 CKAN_LANG_IS_DEFAULT: True 
| 1633360469489 HTTP_ACCEPT: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9' 
| 1633360469489 HTTP_ACCEPT_ENCODING: 'gzip, deflate, br' 
| 1633360469489 HTTP_ACCEPT_LANGUAGE: 'en-US,en;q=0.9' 
| 1633360469489 HTTP_COOKIE: '_ga=GA1.2.960540053.1622127109; _hjid=c51a9257-bb4d-419d-9ea7-ad046e4775de; _ga=GA1.3.960540053.1622127109; kobonaut=3l01cq5czhjpbtx18qf2gb7lc64ashlw; _gcl_au=1.1.1487222393.1633358516; _gid=GA1.2.410090620.1633358516; _gid=GA1.3.410090620.1633358516; auth_tkt="e6cbca9e746d98423f4ee2b8f192b841615b12b62%3Durn%253Aoasis%253Anames%253Atc%253ASAML%253A1.1%253Anameid-format%253AemailAddress%2C4%3Dvazquez%2540unhcr.org!"; auth_tkt="e6cbca9e746d98423f4ee2b8f192b841615b12b62%3Durn%253Aoasis%253Anames%253Atc%253ASAML%253A1.1%253Anameid-format%253AemailAddress%2C4%3Dvazquez%2540unhcr.org!"; ckan=431e58b41a7475b60c44951ecce0a766e7cf4cc7628e889f09fe4a9c85b48d4711282dd2' 
| 1633360469489 HTTP_HOST: 'ridl.unhcr.org' 
| 1633360469489 HTTP_REFERER: 'https://github.com/okfn/ckanext-unhcr/issues/712' 
| 1633360469489 HTTP_SEC_CH_UA: '"Chromium";v="94", "Google Chrome";v="94", ";Not A Brand";v="99"' 
| 1633360469489 HTTP_SEC_CH_UA_MOBILE: '?0' 
| 1633360469489 HTTP_SEC_CH_UA_PLATFORM: '"Linux"' 
| 1633360469489 HTTP_SEC_FETCH_DEST: 'document' 
| 1633360469489 HTTP_SEC_FETCH_MODE: 'navigate' 
| 1633360469489 HTTP_SEC_FETCH_SITE: 'cross-site' 
| 1633360469489 HTTP_SEC_FETCH_USER: '?1' 
| 1633360469489 HTTP_UPGRADE_INSECURE_REQUESTS: '1' 
| 1633360469489 HTTP_USER_AGENT: 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.61 Safari/537.36' 
| 1633360469489 HTTP_X_AMZN_TRACE_ID: 'Root=1-615b1a54-3a61273621096a490104ddf1' 
| 1633360469489 HTTP_X_FORWARDED_FOR: '190.11.151.7' 
| 1633360469489 HTTP_X_FORWARDED_PORT: '443' 
| 1633360469489 HTTP_X_FORWARDED_PROTO: 'https' 
| 1633360469489 PATH_INFO: '/data-container/innovation-service' 
| 1633360469489 REMOTE_ADDR: '10.7.3.178' 
| 1633360469489 REMOTE_PORT: '39653' 
| 1633360469489 REMOTE_USER: '2=urn%3Aoasis%3Anames%3Atc%3ASAML%3A1.1%3Anameid-format%3AemailAddress,4=vazquez%40unhcr.org' 
| 1633360469489 REMOTE_USER_TOKENS: [''] 
| 1633360469489 REQUEST_METHOD: 'GET' 
| 1633360469489 REQUEST_URI: '/data-container/innovation-service' 
| 1633360469489 SERVER_NAME: 'ip-10-7-3-5.eu-west-1.compute.internal' 
| 1633360469489 SERVER_PORT: '5000' 
| 1633360469489 SERVER_PROTOCOL: 'HTTP/1.1' 
| 1633360469489 UWSGI_ROUTER: 'http' 
| 1633360469489 WSGI Variables 
| 1633360469489 -------------- 
| 1633360469489 application: <fanstatic.publisher.Delegator object at 0x7f1c30beaed0> 
| 1633360469489 beaker.cache: <beaker.cache.CacheManager object at 0x7f1c30bead50> 
| 1633360469489 beaker.get_session: <bound method SessionMiddleware._get_session of <beaker.middleware.SessionMiddleware object at 0x7f1c30beac90>> 
| 1633360469489 beaker.session: {'_accessed_time': 1633360469.47389, '_creation_time': 1633358520.594467} 
| 1633360469489 ckan.app: 'pylons_app' 
| 1633360469489 fanstatic.needed: <fanstatic.core.NeededResources object at 0x7f1c30dbdb90> 
| 1633360469489 paste.registry: <paste.registry.Registry object at 0x7f1c30e8aed0> 
| 1633360469489 paste.throw_errors: True 
| 1633360469489 pylons.action_method: <bound method ExtendedOrganizationController.read of <ckanext.unhcr.controllers.extended_organization.ExtendedOrganizationController object at 0x7f1c30db0a50>> 
| 1633360469489 pylons.controller: <ckanext.unhcr.controllers.extended_organization.ExtendedOrganizationController object at 0x7f1c30db0a50> 
| 1633360469489 pylons.environ_config: {'session': 'beaker.session', 'cache': 'beaker.cache'} 
| 1633360469489 pylons.pylons: <pylons.util.PylonsContext object at 0x7f1c30db04d0> 
| 1633360469489 pylons.routes_dict: {'action': u'read', 'controller': u'ckanext.unhcr.controllers.extended_organization:ExtendedOrganizationController', 'id': u'innovation-service'} 
| 1633360469489 repoze.who.api: <repoze.who.api.API object at 0x7f1c30e8ac10> 
| 1633360469489 repoze.who.identity: <repoze.who identity (hidden, dict-like) at 139759069091696> 
| 1633360469489 repoze.who.logger: <logging.Logger object at 0x7f1c30c03350> 
| 1633360469489 repoze.who.plugins: {'saml2auth': 
| 1633360469489 routes.route: <routes.route.Route object at 0x7f1c31c181d0> 
| 1633360469489 routes.url: <routes.util.URLGenerator object at 0x7f1c310b02d0> 
| 1633360469489 uwsgi.node: 'ip-10-7-3-5.eu-west-1.compute.internal' 
| 1633360469489 uwsgi.version: '2.0.19.1' 
| 1633360469489 webob._parsed_query_vars: (GET([]), '') 
| 1633360469489 webob.adhoc_attrs: {'response': <Response at 0x7f1c30db0390 200 OK>, 'language': 'en-us'} 
| 1633360469489 wsgi process: 'Multiprocess' 
| 1633360469489 wsgi.file_wrapper: <built-in function uwsgi_sendfile> 
| 1633360469489 wsgiorg.routing_args: (<routes.util.URLGenerator object at 0x7f1c310b02d0>, {'action': u'read', 'controller': u'ckanext.unhcr.controllers.extended_organization:ExtendedOrganizationController', 'id': u'innovation-service'}) 
| 1633360469489 ------------------------------------------------------------ 
| 1633360469577 2021-10-04 15:14:29,576 INFO [ckan.lib.base] /data-container/innovation-service render time 0.090 seconds 
--------------------------------------------
<pre>

**Comment by avdata99** [comment]
Probably related with the name
IDs:
 - 5bec4da1-0d4b-4f33-ab3b-7b7c9f539d5b innovation-service
 - 534b78f1-9c39-4678-a6ff-e558be5e83d0 innovation_service

Access those containers using IDs is also not working

**Comment by amercader** [comment]
I've manually fixed the data container in prod (the edit page worked): https://ridl.unhcr.org/data-container/innovation_service

This seems to be a recursion problem with the ckanext-hierarchy extension. Somehow this data container ended up self-referencing itself. It's really hard to debug without following the exact steps that the users followed to create them. So I suggest park it until we noticed that again.
